### PR TITLE
Fix bem-suggest for bem-xjst 8

### PR DESCRIPTION
**Pull request by d3m1d0v**
Using suggest in bemtree with bem-xjst8 it working incorrectly. This change fix it.

### CODE REVIEW

File: blocks/suggest/suggest.bemhtml.js [hunk 1]
```diff
@@ -3,7 +3,12 @@ block('suggest')(
         return applyNext({ _suggest : this.ctx });
     }),
 
-    js()(true),
+    js()(function() {
+        var prevJs = applyNext(),
```
Comment (tadatuta) [inline blocks/suggest/suggest.bemhtml.js]: it's enough just
```js
return applyNext() || true;
```

Comment (d3m1d0v) [reply blocks/suggest/suggest.bemhtml.js]: But if I use 
```js
js()(function() {
        return applyNext() || true;
})
```
js-parameters don't added to data-bem content for suggest block

Comment (belozer) [reply blocks/suggest/suggest.bemhtml.js]: @d3m1d0v are you sure? https://goo.gl/5cLptw

Comment (d3m1d0v) [reply blocks/suggest/suggest.bemhtml.js]: @belozer Yes, and  when suggest focused it not initializing (no added `suggest_js_inited` class)

Comment (belozer) [reply blocks/suggest/suggest.bemhtml.js]: @d3m1d0v please show your html. In demo all ok.

**Comment by d3m1d0v** [comment]
@tadatuta @belozer

File: blocks/suggest/_has-dataprovider/suggest_has-dataprovider.bemhtml.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
...
-    var prevJs = applyNext();
-    return prevJs && prevJs !== true? this.extend(prevJs, this.ctx.dataprovider) : this.ctx.dataprovider;
-})
+block('suggest')
+    .match(function() {
+        return this.mods['has-dataprovider'];
+    })
+    .js()(function() {
+        var prevJs = applyNext();
+        return prevJs && prevJs !== true? this.extend(prevJs, this.ctx.dataprovider) : this.ctx.dataprovider || true;
```
Comment (belozer) [inline blocks/suggest/_has-dataprovider/suggest_has-dataprovider.bemhtml.js]: Хм... Кажется это место можно упросить до такой записи 
```js
return this.extend(applyNext() || {}, this.ctx.dataprovider);
```

Comment (d3m1d0v) [reply blocks/suggest/_has-dataprovider/suggest_has-dataprovider.bemhtml.js]: @belozer Да, обновил.

**Pull request merged**